### PR TITLE
[ADD] beesdoo_portal_shift: Show available shift

### PR DESCRIPTION
**Pull request by remytms**
For review.

Don't merge !

### CODE REVIEW

File: beesdoo_portal_shift/controllers/main.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# -*- coding: utf8 -*-
+from datetime import datetime
+from math import floor
+
+from openerp import http
+from openerp.http import request
+
+class Sandbox(http.Controller):
```
Comment (houssine78) [inline beesdoo_portal_shift/controllers/main.py]: change for a meaningful class name

File: beesdoo_portal_shift/controllers/main.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        shift_templates = []
+        current_template = None
+        current_shift_template = None
+        current_remaining_space = 0
+        for shift in shifts:
+            # For a planning id, count the number of shift that don't
+            # have a worker.
+            if shift.task_template_id == current_template:
+                # If we are in the same template then update the number
+                # of available sapce
```
Comment (houssine78) [inline beesdoo_portal_shift/controllers/main.py]: typo on available "space"

Comment (houssine78) [reply beesdoo_portal_shift/controllers/main.py]: put place instead

Comment (remytms) [reply beesdoo_portal_shift/controllers/main.py]: Ok, I see my typo error. But I think "palce" is for a "location". English speaker use "space" or "room" to say "spot occupied by someone".

File: beesdoo_portal_shift/controllers/main.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+class Sandbox(http.Controller):
+
+    @http.route('/shift_irregular_worker', auth='public', website=True)
+    def shift_irregular_worker(self, **kwargs):
+        # Get all the shifts in the future with no worker
+        now = datetime.now()
+        shifts = request.env['beesdoo.shift.shift'].search(
+            [('start_time', '>', now.strftime("%Y-%m-%d %H:%M:%S")),
+            ('worker_id', '=', False)],
+            order="start_time, task_template_id, task_type_id",
```
Comment (houssine78) [inline beesdoo_portal_shift/controllers/main.py]: what if task_template_id and start_time were inveresed?

File: beesdoo_portal_shift/controllers/main.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        current_shift_template = None
+        current_remaining_space = 0
+        for shift in shifts:
+            # For a planning id, count the number of shift that don't
+            # have a worker.
+            if shift.task_template_id == current_template:
+                # If we are in the same template then update the number
+                # of available sapce
+                current_remaining_space = current_remaining_space + 1
+            else:
```
Comment (houssine78) [inline beesdoo_portal_shift/controllers/main.py]: the else should comes has the main conditional statement

File: beesdoo_portal_shift/data/ir.model.access.csv [hunk 5]
```diff
@@ -0,0 +1,6 @@
+id,name,model_id/id,group_id/id,perm_read,perm_write,perm_create,perm_unlink
+access_beesdoo_shift_template,Public Read Template,beesdoo_shift.model_beesdoo_shift_template,,1,0,0,0
+access_beesdoo_shift_shift,Public Read Shift,beesdoo_shift.model_beesdoo_shift_shift,,1,0,0,0
+access_beesdoo_shift_type,Public Read Type Shift,beesdoo_shift.model_beesdoo_shift_type,,1,0,0,0
+access_beesdoo_shift_planning,Public Read Planning,beesdoo_shift.model_beesdoo_shift_planning,,1,0,0,0
+access_beesdoo_cooperative_status_history,Public Read Cooperative Status History,beesdoo_shift.model_cooperative_status_history,,1,0,0,0
```
Comment (houssine78) [inline beesdoo_portal_shift/data/ir.model.access.csv]: Dont give read access to public user. Instead use sudo() to instanciate function as admin

**Comment by remytms** [comment]
Virginie propose to refactor the code in another PR. But push this one in production, because it's really needed.

@tfrancoi, can you merge this ?

**Comment by vdewulf** [comment]
Tested OK on runbot
http://26007-9-0-s0030prior-remy-dcac59.runbot.bees-coop.be
@tfrancoi could you merge please?

**Comment by remytms** [comment]
@tfrancoi, I changed the name of the module. :)

**Comment by houssine78** [comment]
ok je peux déjà merger cela.

**Comment by houssine78** [comment]
tiesn @tfrancoi je n'ai plus les droits de commit?

**Comment by houssine78** [comment]
ah benh si maintenant je les ai... bizarre

**Comment by houssine78** [comment]
@remytms c'est mergé!

**Pull request merged**